gcloud/storage/batch.py

```diff
@@ -109,22 +159,22 @@ def _do_request(self, method, url, headers, data):
                 and ``content`` (a string).
         :returns: The HTTP response object and the content of the response.
         """
-        if method == 'GET':
-            _req = self._connection.http.request
-            return _req(method=method, uri=url, headers=headers, body=data)
-
         if len(self._requests) >= self._MAX_BATCH_SIZE:
             raise ValueError("Too many deferred requests (max %d)" %
                              self._MAX_BATCH_SIZE)
         self._requests.append((method, url, headers, data))
-        return NoContent(), ''
-
-    def finish(self):
-        """Submit a single `multipart/mixed` request w/ deferred requests.
-
-        :rtype: list of tuples
-        :returns: one ``(status, reason, payload)`` tuple per deferred request.
-        :raises: ValueError if no requests have been deferred.
+        result = _FutureDict()
+        self._futures.append(target_object)
+        if target_object is not None:
+            target_object._properties = result
```

Comment:
Dunno -- I guess I was worried about cases where there might be only a subset of the full property set coming back in a batched request (using `fields` or maybe `projection`).  In that case, I think we would not want to be replacing the `target_object._properties` dict unconditionally at batch-request time.